File: YABA/Core/Model/YabaColor.swift
//
//  YabaColor.swift
//  YABA
//
//  Created by Ali Taha on 13.10.2024.
//
// swiftlint:disable all

import SwiftUI

enum YabaColor: Int, Codable, CaseIterable {
    case none = 0
    case blue = 1
    case brown = 2
    case cyan = 3
    case gray = 4
    case green = 5
    case indigo = 6
    case mint = 7
    case orange = 8
    case pink = 9
    case purple = 10
    case red = 11
    case teal = 12
    case yellow = 13

    func getUIColor() -> Color {
        switch self {
        case .blue:
                .blue
        case .brown:
                .brown
        case .cyan:
                .cyan
        case .gray:
                .gray
        case .green:
                .green
        case .orange:
                .orange
        case .purple:
                .purple
        case .red:
                .red
        case .yellow:
                .yellow
        case .indigo:
                .indigo
        case .mint:
                .mint
        case .teal:
                .teal
        case .pink:
                .pink
        default:
                .accentColor
        }
    }
    
    func getUIText() -> String {
        switch self {
        case .blue:
                "Blue"
        case .brown:
                "Brown"
        case .cyan:
                "Cyan"
        case .gray:
                "Gray"
        case .green:
                "Green"
        case .orange:
                "Orange"
        case .purple:
                "Purple"
        case .red:
                "Red"
        case .yellow:
                "Yellow"
        case .indigo:
                "Indigo"
        case .mint:
                "Mint"
        case .teal:
                "Teal"
        case .pink:
                "Pink"
        default:
                "Theme Color"
        }
    }
}

File: YABA-New/YABA/app/src/main/java/dev/subfly/yaba/core/model/utils/YabaColor.kt
package dev.subfly.yaba.core.model.utils

/**
 * Mirrors Darwin's YabaColor mapping for cross-platform consistency. Persisted as an Int in
 * entities.
 */
enum class YabaColor(val code: Int) {
    NONE(0),
    BLUE(1),
    BROWN(2),
    CYAN(3),
    GRAY(4),
    GREEN(5),
    INDIGO(6),
    MINT(7),
    ORANGE(8),
    PINK(9),
    PURPLE(10),
    RED(11),
    TEAL(12),
    YELLOW(13);

    fun iconTintArgb(): Long =
        when (this) {
            NONE -> 0x00000000L
            BLUE -> 0xFF0088FF
            BROWN -> 0xFFAC7F5E
            CYAN -> 0xFF00C0E8
            GRAY -> 0xFF8E8E93
            GREEN -> 0xFF34C759
            INDIGO -> 0xFF6155F5
            MINT -> 0xFF00C8B3
            ORANGE -> 0xFFFF8D28
            PINK -> 0xFFFF2D55
            PURPLE -> 0xFFCB30E0
            RED -> 0xFFFF383C
            TEAL -> 0xFF00C3D0
            YELLOW -> 0xFFFFCC00
        }

    companion object {
        fun fromCode(code: Int): YabaColor = entries.firstOrNull { it.code == code } ?: NONE

        fun fromRoleString(raw: String?): YabaColor {
            val normalized = raw?.trim()?.lowercase().orEmpty()
            if (normalized.isBlank()) return NONE
            if (normalized == "default") return NONE
            return entries.firstOrNull { it.name.lowercase() == normalized } ?: NONE
        }
    }
}
